retrievalmarket/impl/clientstates/client_states.go

```diff
@@ -25,7 +26,7 @@ type ClientDealEnvironment interface {
 	OpenDataTransfer(ctx context.Context, to peer.ID, proposal *rm.DealProposal, legacy bool) (datatransfer.ChannelID, error)
 	SendDataTransferVoucher(context.Context, datatransfer.ChannelID, *rm.DealPayment, bool) error
 	CloseDataTransfer(context.Context, datatransfer.ChannelID) error
-	FinalizeBlockstore(context.Context, rm.DealID) error
+	FinalizeBlockstore(context.Context, cid.Cid) error
```

Comment:
This is used with a PayloadCID. I don't think we should use it with a PayloadCID because there could be more than one retrieval in progress (from different miners) for the same payload CID.